app/browser/reducers/sitesReducer.js

```diff
@@ -54,35 +54,57 @@ const sitesReducer = (state, action, immutableAction) => {
         break
       }
     case appConstants.APP_ADD_SITE:
-      const isSyncEnabled = syncEnabled()
-      if (Immutable.List.isList(action.siteDetail)) {
-        action.siteDetail.forEach((s) => {
-          state = siteUtil.addSite(state, s, action.tag, undefined, action.skipSync)
+      {
+        const isSyncEnabled = syncEnabled()
+        if (Immutable.List.isList(action.siteDetail)) {
+          action.siteDetail.forEach((s) => {
+            state = siteUtil.addSite(state, s, action.tag, action.skipSync)
+            if (isSyncEnabled) {
+              state = syncUtil.updateSiteCache(state, s)
+            }
+          })
+        } else {
+          let sites = state.get('sites')
+          if (!action.siteDetail.get('folderId') && siteUtil.isFolder(action.siteDetail)) {
+            action.siteDetail = action.siteDetail.set('folderId', siteUtil.getNextFolderId(sites))
+          }
+          state = siteUtil.addSite(state, action.siteDetail, action.tag, action.skipSync)
           if (isSyncEnabled) {
-            state = syncUtil.updateSiteCache(state, s)
+            state = syncUtil.updateSiteCache(state, action.siteDetail)
           }
-        })
-      } else {
-        let sites = state.get('sites')
-        if (!action.siteDetail.get('folderId') && siteUtil.isFolder(action.siteDetail)) {
-          action.siteDetail = action.siteDetail.set('folderId', siteUtil.getNextFolderId(sites))
         }
-        state = siteUtil.addSite(state, action.siteDetail, action.tag, action.originalSiteDetail, action.skipSync)
-        if (action.destinationDetail) {
-          const sourceKey = siteUtil.getSiteKey(action.siteDetail)
-          const destinationKey = siteUtil.getSiteKey(action.destinationDetail)
+        break
+      }
+    case appConstants.APP_ADD_BOOKMARK:
+    case appConstants.APP_EDIT_BOOKMARK:
+      {
+        const isSyncEnabled = syncEnabled()
```

Comment:
nitpick for this section- it has enough logic and branches (if/else) that it would be nice to pull it out to a function so that it can be easily tested